claat/fetch.go

```diff
@@ -227,22 +228,42 @@ func fetchDriveFile(id string, nometa bool) (*resource, error) {
 
 var crcTable = crc64.MakeTable(crc64.ECMA)
 
-func slurpBytes(client *http.Client, dir, url string, n int) (string, error) {
-	res, err := retryGet(client, url, n)
+func slurpBytes(client *http.Client, codelabSrc, dir, imgURL string, n int) (string, error) {
+	// images can be local in Markdown cases or remote.
+	// Only proceed a simple copy on local reference.
+	var b []byte
+	var ext string
+	u, err := url.Parse(imgURL)
 	if err != nil {
 		return "", err
 	}
-	defer res.Body.Close()
-	b, err := ioutil.ReadAll(res.Body)
+	if u.Host == "" {
+		imgURL = path.Join(path.Dir(codelabSrc), imgURL)
```

Comment:
I think that restricting elements to a subpath from the codelab directory is a nice way to mitigate those issues.

However, prepending a / to imgURL will not achieve this (I just tried and checked golang stdlib): `path.Join` is cleaning each elements of the `Join` (apart from the first one), and thus, strips any prepended or appended /. Thus, `path.Join("bar", "/../foo")` is equivalent to `path.Join("bar", "../foo")`.
I'm going to use some absolute path method to ensure we capture the asset as being in the tree of the codelab directory.